tools/buildmgr/cbuildgen/config/GCC.11.2.1.cmake

```diff
@@ -15,6 +15,77 @@ set(CC ${TOOLCHAIN_ROOT}/${PREFIX}gcc${EXT})
 set(CXX ${TOOLCHAIN_ROOT}/${PREFIX}g++${EXT})
 set(OC ${TOOLCHAIN_ROOT}/${PREFIX}objcopy${EXT})
 
+# Helpers
+
+function(cbuild_set_defines lang defines)
+  set(TMP_DEFINES)
+  foreach(DEFINE ${${defines}})
+    string(REPLACE "\"" "\\\"" ENTRY ${DEFINE})
+    string(REGEX REPLACE "=.*" "" KEY ${ENTRY})
+    if (KEY STREQUAL ENTRY)
+      set(VALUE "1")
+    else()
+      string(REGEX REPLACE ".*=" "" VALUE ${ENTRY})
+    endif()
+    if(${lang} STREQUAL "AS_LEG")
+      string(APPEND TMP_DEFINES "--defsym ${KEY}=${VALUE} ")
+    elseif(${lang} STREQUAL "AS_GNU")
+      string(APPEND TMP_DEFINES "-Wa,-defsym,\"${KEY}=${VALUE}\" ")
+    else()
+      string(APPEND TMP_DEFINES "-D${ENTRY} ")
+    endif()
+  endforeach()
+  set(${defines} ${TMP_DEFINES} PARENT_SCOPE)
+endfunction()
+
+set(OPTIMIZE_VALUES       "debug" "none" "balanced" "size" "speed")
+set(OPTIMIZE_CC_FLAGS     "-Og"   "-O0"  "-O2"      "-Oz" "-O3")
+set(OPTIMIZE_AS_GNU_FLAGS ${OPTIMIZE_CC_FLAGS})
+set(OPTIMIZE_ASM_FLAGS    ${OPTIMIZE_CC_FLAGS})
+set(OPTIMIZE_CXX_FLAGS    ${OPTIMIZE_CC_FLAGS})
+set(OPTIMIZE_LD_FLAGS     ${OPTIMIZE_CC_FLAGS})
+
+set(DEBUG_VALUES          "on"      "off")
+set(DEBUG_CC_FLAGS        "-g3"     "-g0")
+set(DEBUG_CXX_FLAGS       ${DEBUG_CC_FLAGS})
+set(DEBUG_LD_FLAGS        ${DEBUG_CC_FLAGS})
+set(DEBUG_AS_GNU_FLAGS    ${DEBUG_CC_FLAGS})
+set(DEBUG_ASM_FLAGS       ${DEBUG_CC_FLAGS})
+
+set(WARNINGS_VALUES       "on"     "off")
+set(WARNINGS_AS_LEG_FLAGS "--warn" "--no-warn")
+set(WARNINGS_CC_FLAGS     ""       "-w")
+set(WARNINGS_ASM_FLAGS    ${WARNINGS_CC_FLAGS})
+set(WARNINGS_AS_GNU_FLAGS ${WARNINGS_CC_FLAGS})
+set(WARNINGS_CXX_FLAGS    ${WARNINGS_CC_FLAGS})
+set(WARNINGS_LD_FLAGS     ${WARNINGS_CC_FLAGS})
+
+function(cbuild_set_option_flags lang option value flags)
+  if(NOT DEFINED ${option}_${lang}_FLAGS)
+    return()
+  endif()
+  list(FIND ${option}_VALUES "${value}" _index)
+  if (${_index} GREATER -1)
+    list(GET ${option}_${lang}_FLAGS ${_index} flag)
+    set(${flags} "${flag} ${${flags}}" PARENT_SCOPE)
+  else()
+    string(TOLOWER "${option}" _option)
+    message(FATAL_ERROR "unkown '${_option}' value '${value}' !")
+  endif()
+endfunction()
+
+function(cbuild_set_options_flags lang optimize debug warnings flags)
+  set(opt_flags)
+  # GCC provide a better optimization level for debug
```

Comment:
is it fine to have this tweek, or can just have `-O0`

Reply:
In my opinion it is fine but it should not overwrite an user selection. In other words I would check whether `optimize` is empty instead of `none`.

Reply:
your proposal seems better